modules/otel-collector-deployment-modes.adoc

```diff
@@ -0,0 +1,49 @@
+//Module included in the following assemblies:
+//
+// * observability/otel/otel-collector/otel-collector-configuration-intro.adoc
+
+:_mod-docs-content-type: REFERENCE
+[id="otel-collector-deployment-modes_{context}"]
+= Deployment modes
+
+The OpenTelemetry Collector supports the following deployment modes:
+
+* Using the relevant Kubernetes workload resource:
++
+Deployment:: The default.
++
+StatefulSet:: If you need to run stateful workloads, for example when using the Collector's File Storage Extension or Tail Sampling Processor, you can choose the StatefulSet deployment mode.
```

Comment:
Instead of `you can choose`, it'd say `it is recommended`

Reply:
Thank you for the suggestion, Andreas, I avoided `recommended` because of style guidance: https://redhat-documentation.github.io/supplementary-style-guide/#recommend